### DISCUSSION TITLE
Please add a way to start a spigot server w/out downloading the .jar file if it already exists locally

### DISCUSSION THREAD

**Discussion by mrfuntime** [itzg/docker-minecraft-server#2343] [Ideas]
I have my docker compose file hard coded to run spigot version 1.18.2 and, once I've run the docker once, that .jar file is resident locally in my image.  Since I already have it at that point, there's no need to download it again.

Normally, downloading an image one already has isn't an issue other than wasting bandwidth, etc., but sometime on 08/16/2023 the download.getbukkit.org site (where the image is coded to pull Spigot .jar files from) has begun throwing code 522 errors.  The end result is that the image keeps trying to download the .jar file and never starts the actual minecraft server even though I already have the .jar file locally.

I'd appreciate it if there was a way for the image to check for this condition and bypass the download if it's not needed.

I may look into doing this myself if time and circumstance permit but I'm an entry level coder and it will likely take me longer than it would someone who is already seasoned.

Thank you in advance for your consideration.  This really is the best MC docker I've found out there and, notwithstanding the above, I have zero complaints with it.

- M

**Comment by itzg** [comment]
That's odd, `curl` is being told not to redownload the file but perhaps getbukkit isn't compliant

https://github.com/itzg/docker-minecraft-server/blob/master/scripts/start-deployBukkitSpigot#L78

Side note: Perhaps you could switch to Paper since it is compatible with Spigot and Bukkit plugins

https://docs.papermc.io/paper/adding-plugins

I'll queue this up for investigating the Spigot download, but yes any research and PRs are welcome. Setting `DEBUG` env var to "true" and investigating those logs could help.

**Reply by mrfuntime** [reply]
Thanks for the quick response.  I won't be able to interact with it til I get home this evening but I'll definitely entertain trying paper; and/or going the debug route to get more info for you.  Thanks again.
